### PR TITLE
Expose `LineEdit` `edit` and `unedit` methods.

### LINKED ISSUE #88235: Color Picker on Mobile causes keyboard to popup by default
### Tested versions

v4.3.dev2.official [352434668], likely has existed for longer.

### System information

Pixel 6a Android

### Issue description

When clicking on the color picker button it by default brings up the onscreen keyboard. Likely because the 'Hex' field is active by default.

Why is this a bug? See below:
Screenshot_20240212-173731

Setting hex_visible to false does not work around the issue.

### Steps to reproduce

Add a ColorPickerButton node.
Click it.

### Minimal reproduction project (MRP)

Am not near computer, but can MRP if required later.

### LINKED ISSUE #97088: `LineEdit` prevents itself from entering text after pressing enter.
### Tested versions

Reproducible in: v4.4.dev.custom_build [6681f2563]
Not Reproducible in: v4.4.dev2.official [97ef3c837]

### System information

Windows 10 - Godot v4.4.dev.custom_build [6681f2563] - OpenGL (Compatibility) - Integrated

### Issue description

After [#87674](https://github.com/godotengine/godot/pull/87674), `LineEdit` needs to be pressed twice with enter before text could be entered again. 

This impacts usability with a developer console addon I've been using and there's no available function or variable now to toggle into edit mode from the instanced `LineEdit`.

### Steps to reproduce

Play the MRP
Input Text
Submit by pressing enter
`LineEdit` is now unresponsive until enter is pressed for a second time.

### Minimal reproduction project (MRP)

[lineedit_console.zip](https://github.com/user-attachments/files/17017814/lineedit_console.zip)

### LINKED ISSUE #97096: `ScriptEditorPlugin` search box requires pressing `Enter` twice to find next search.
### Tested versions

4.4.dev

### System information

Windows 10

### Issue description

see https://github.com/godotengine/godot/issues/97088#issuecomment-2354186925

### Steps to reproduce

see https://github.com/godotengine/godot/issues/97088#issuecomment-2354186925

### Minimal reproduction project (MRP)

N/A

### LINKED ISSUE #97737: `LineEdit` selects all text when it's just focused with arrow keys.
### Tested versions

4.4.dev

### System information

Windows 10

### Issue description

When focusing `LineEdit` with arrow keys it selects all text instead of selecting all text only when editing.

### Steps to reproduce

Focus any `SpinBox` with arrow keys.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by WhalesState**
* Fixes: https://github.com/godotengine/godot/issues/97088
* Fixes: https://github.com/godotengine/godot/issues/97096
* Fixes: https://github.com/godotengine/godot/issues/97737
* Fixes: https://github.com/godotengine/godot/issues/88235

### CODE REVIEW

File: doc/classes/LineEdit.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 		- When the [LineEdit] control is focused using the keyboard arrow keys, it will only gain focus and not enter edit mode.
 		- To enter edit mode, click on the control with the mouse or press the "ui_text_submit" action (default: [kbd]Enter[/kbd] or [kbd]Kp Enter[/kbd]).
 		- To exit edit mode, press "ui_text_submit" or "ui_cancel" (default: [kbd]Escape[/kbd]) actions.
-		- Check [method is_editing] and [signal editing_toggled] for more information.
+		- Check [method edit], [method unedit], [method is_editing] and [signal editing_toggled] for more information.
```
Comment (akien-mga) [inline doc/classes/LineEdit.xml:11]: Oxford comma:
```suggestion
		- Check [method edit], [method unedit], [method is_editing], and [signal editing_toggled] for more information.
```

File: doc/classes/LineEdit.xml [hunk 2]
```diff
@@ -211,6 +218,12 @@
 				Selects the whole [String].
 			</description>
 		</method>
+		<method name="unedit">
+			<return type="void" />
+			<description>
+				Allows exiting edit mode and preserve focus.
```
Comment (akien-mga) [inline doc/classes/LineEdit.xml:224]:
```suggestion
				Allows exiting edit mode while preserving focus.
```

**Comment by WhalesState** [comment]
Feel free to change the title, since all the fixes are connected to the new behavior.

**Comment by WhalesState** [comment]
Regression! pressing `escape` key to hide the `ColorPicker` window will revert back to the old color 😞

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**